ios/GenericShare.m

```diff
@@ -9,17 +9,26 @@
 #import "GenericShare.h"
 
 @implementation GenericShare
-    RCT_EXPORT_MODULE();
 - (void)shareSingle:(NSDictionary *)options
     failureCallback:(RCTResponseErrorBlock)failureCallback
     successCallback:(RCTResponseSenderBlock)successCallback
     serviceType:(NSString*)serviceType {
-
+    
+    static BOOL canOpenViaApp;
+    
+    canOpenViaApp = [SLComposeViewController isAvailableForServiceType:serviceType];
+    
+    if ([options[@"social"] isEqualToString:@"facebook"]){
+        canOpenViaApp = [self isAppInstalled:@"fbauth2"];
```

Comment:
This should be something like `canOpenViaApp = canOpenViaApp || [self isAppInstalled:@"fbauth2"];` since for pre-iOS 11 systems, isAvailableForServiceType should presumably work correctly and so we don't want to accidentally trigger false negatives. But test it and make sure.

Same situation with twitter.

Reply:
tested on 10.3.3 current implementation is working no need for the OR.